src/inputSystem.ml

```diff
@@ -990,6 +992,82 @@ function
 
   | Horn _ -> raise (UnsupportedFileFormat "Horn")
 
+let get_bv_sizes (type s) : s t -> IntSet.t = 
+  let over_svar = (fun acc svar ->
+    let ty = SVar.type_of_state_var svar in 
+    match Type.node_of_type ty with 
+    | Type.BV width -> IntSet.add width acc
+    | _ -> acc  
+  ) in
+  function 
+  | Lustre (main_subs, globals, _) -> 
+    let subsystems = S.all_subsystems_of_list main_subs in 
+    let sources = List.map (fun subsys -> subsys.S.source) subsystems in 
+    (* Get sizes from every Lustre node *)
+    let sizes = List.fold_left (fun acc source -> 
+      (* Oracle sizes *)
+      let acc = List.fold_left over_svar acc source.N.oracles in 
+      (* Input sizes *)
+      let acc = List.fold_left over_svar acc (LustreIndex.values source.N.inputs) in 
+      (* Output sizes *)
+      let acc = List.fold_left over_svar acc (LustreIndex.values source.N.outputs) in 
+      (* Local sizes *)
+      let acc = List.fold_left over_svar acc (Lib.concat_map LustreIndex.values source.N.locals) in 
+      (* Global sizes *)
+      let acc = List.fold_left over_svar acc (SVar.StateVarHashtbl.to_seq_keys globals.state_var_bounds |> List.of_seq) in 
+      acc
+    ) IntSet.empty sources in 
+    sizes
+  | Moxi checks -> 
+    let subsystems = List.map fst checks in 
+    let sources = List.map (fun subsys -> subsys.S.source) subsystems in 
+    let state_vars = Lib.concat_map TransSys.state_vars sources in 
+    List.fold_left over_svar IntSet.empty state_vars
+  | Native sub -> 
+    let subsystems = S.all_subsystems sub in
+    let sources = List.map (fun subsys -> subsys.S.source) subsystems in 
+    let state_vars = Lib.concat_map TransSys.state_vars sources in 
+    List.fold_left over_svar IntSet.empty state_vars
+  | Horn _ -> IntSet.empty
```

Comment:
This code could be factored out. Could you create an auxiliary function parameterized with `over_svar`, and reduce `get_bv_sizes` and `get_ubv_sizes` to a call to this new function?